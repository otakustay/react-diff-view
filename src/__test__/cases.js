/* In case you are looking for type of hunk, see src/Hunk/index.js */
import {parseDiff} from '..';

export const basic = `diff --git a/src/__test__/index.test.jsx b/src/__test__/index.test.jsx
index 643c2f0..7883597 100644
--- a/src/__test__/index.test.jsx
+++ b/src/__test__/index.test.jsx
@@ -21,3 +21,3 @@ describe('basic test', () => {
     test('App renders correctly', () => {
-        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();
+        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();
     });`;

export const basicHunk = parseDiff(basic)[0].hunks[0];

export const multiple = `diff --git a/src/index1.js b/src/index1.js
index e69de29..d00491f 100644
--- a/src/index1.js
+++ b/src/index1.js
@@ -0,0 +1 @@
+1
diff --git a/src/index.js b/src/index2.js
index d00491f..0cfbf08 100644
--- a/src/index2.js
+++ b/src/index2.js
@@ -1 +1 @@
-1
+2
diff --git a/src/index3.js b/src/index3.js
index 0cfbf08..e69de29 100644
--- a/src/index3.js
+++ b/src/index3.js
@@ -1 +0,0 @@
-2`;

export const multipleHunk = parseDiff(basic)[0].hunks[0];
