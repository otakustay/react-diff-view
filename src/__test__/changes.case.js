
export const normalChange = {isNormal: true, lineNumber: 0, oldLineNumber: 0};
export const insertChange = {isInsert: true, lineNumber: 0, oldLineNumber: 0};
export const deleteChange = {isDelete: true, lineNumber: 0, oldLineNumber: 0};
