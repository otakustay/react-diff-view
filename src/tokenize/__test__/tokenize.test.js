import {tokenize} from '..';
import {basicHunk, multipleHunk} from '../../__test__/cases';

describe('tokenize', () => {
    test('basic', () => {
        expect(tokenize([], {})).toEqual({
            new: [[{type: 'text', value: ''}]],
            old: [[{type: 'text', value: ''}]]
        });
    });

    test('different config', () => {
        expect(tokenize([], {
            highlight: true,
            refractor: {highlight: text => [{type: 'text', value: text}]}
        })).toEqual({
            new: [[{type: 'text', value: ''}]],
            old: [[{type: 'text', value: ''}]]
        });

        expect(tokenize([], {oldSource: 'A'})).toEqual({
            new: [[{type: 'text', value: 'A'}]],
            old: [[{type: 'text', value: 'A'}]]
        });
    });

    test('withHunk', () => {
        expect(tokenize([basicHunk, multipleHunk], {})).toMatchSnapshot();
    });
});
