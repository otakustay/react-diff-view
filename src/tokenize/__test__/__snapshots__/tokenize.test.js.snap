// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tokenize withHunk 1`] = `
Object {
  "new": Array [
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "    test('App renders correctly', () => {",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "        expect(renderer.create(<App diffText={'diff'} />).toJSON()).toMatchSnapshot();",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "    });",
      },
    ],
  ],
  "old": Array [
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "    test('App renders correctly', () => {",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "        expect(renderer.create(<App diffText={'deff'} />).toJSON()).toMatchSnapshot();",
      },
    ],
    Array [
      Object {
        "type": "text",
        "value": "    });",
      },
    ],
  ],
}
`;
